package.json

```diff
@@ -54,6 +54,8 @@
     "babel-polyfill": "^6.23.0",
     "babel-preset-env": "~1.4.0",
     "babel-preset-react": "^6.24.1",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-stage-1": "^6.24.1",
```

Comment:
are those required?

Reply:
```
const renderIcon = () => <i className={icon} />;
```
Yes. W/o those 2 lines this code would not compile ^^^

Reply:
what was the error? stage-1 should normally not be required.

Reply:
```
    at transpile (/home/martin/Projects/manageiq-ui-classic/node_modules/babel-loader/lib/index.js:63:13)
    at Object.module.exports (/home/martin/Projects/manageiq-ui-classic/node_modules/babel-loader/lib/index.js:163:20)
 @ ./app/javascript/react/textual_summary/textual_group.jsx 20:17-39
 @ ./app/javascript/react/textual_summary/textual_row.jsx
 @ ./app/javascript/react/textual_summary/index.jsx
 @ ./app/javascript/packs/application-common.js

ERROR in ./app/javascript/react/textual_summary/icon_or_image.jsx
Module build failed: SyntaxError: Unexpected token (12:13)

  10 |   }
  11 | 
> 12 |   renderIcon = () => <i className={this.props.icon} title={this.props.title} />;
     |              ^

```

```
diff --git a/.babelrc b/.babelrc
index 9d3ea8b..936c5c8 100644
--- a/.babelrc
+++ b/.babelrc
@@ -1,5 +1,6 @@
 {
-  "presets": [["env", { "modules": false }], "react", "es2015", "stage-1"],
+  //"presets": [["env", { "modules": false }], "react", "es2015", "stage-1"],
+  "presets": [["env", { "modules": false }], "react", "es2015"],
   "env": {
     "test": {
       "plugins": ["transform-es2015-modules-commonjs"]
```

Reply:
The problem is it's inside the class, so I need class fields. I can change the code to use methods instead or make it a function, not a class.

Reply:
The missing feature is called class fields: https://github.com/tc39/proposal-class-fields

```
class Foo {
  renderIcon = 5;
}
```

Currently at stage3 ..  so maybe we just need to bump es2015 to a newer version..

Reply:
Looks like maybe we should just be using `env` (and `react`) with some settings to target IE9+.

Reply:
We need to figure the best way ;-)

Figuring the best way, I have some notes:

I like some of the poweful expressivity of the functional notation. But there are some parts that are really ugly inside JSX:
 * the contional inclusion of markup: `(condition && (<your_crap_here>))`
```
            {(index === 0) && (                                                                     
              <td rowSpan={item.value.length} className="label">{item.label}</td>                   
            )}   
```

 * case construct (hash + lookup): 
```
      {{                                                                                            
          'shared/summary/textual': (                                                                 
            <GenericGroup items={props.group.items} title={props.group.title} />                      
          ),                                                                                          
          'shared/summary/textual_tags': (                                                            
            <TagGroup items={props.group.items} title={props.group.title} />                          
          ),                                                                                          
          'shared/summary/textual_table': (                                                           
>>          <SimpleTable items={props.group.items} title={props.group.title} />                       
          ),                                                                                          
        }[props.group.partial]} 
```

 * I am glad for the `<React.Fragment>`. That helps.

Overall the JSX seems to be just another way to write markup and not particularly better than HAML. How many do we need?